plugin/trino-memsql/src/test/java/io/trino/plugin/memsql/TestMemSqlTypeMapping.java

```diff
@@ -463,40 +460,35 @@ public void testDate()
 
         ZoneId someZone = ZoneId.of("Europe/Vilnius");
 
-        for (String timeZoneId : ImmutableList.of(UTC_KEY.getId(), jvmZone.getId(), someZone.getId())) {
-            Session session = Session.builder(getSession())
-                    .setTimeZoneKey(TimeZoneKey.getTimeZoneKey(timeZoneId))
-                    .build();
-            dateTestCases(memSqlDateDataType(value -> formatStringLiteral(value.toString())), jvmZone, someZone)
-                    .execute(getQueryRunner(), session, memSqlCreateAndInsert("tpch.test_date"));
-            dateTestCases(dateDataType(), jvmZone, someZone)
-                    .execute(getQueryRunner(), session, trinoCreateAsSelect(session, "test_date"));
-            dateTestCases(dateDataType(), jvmZone, someZone)
-                    .execute(getQueryRunner(), session, trinoCreateAsSelect(getSession(), "test_date"));
-            dateTestCases(dateDataType(), jvmZone, someZone)
-                    .execute(getQueryRunner(), session, trinoCreateAndInsert(session, "test_date"));
-        }
-    }
-
-    private DataTypeTest dateTestCases(DataType<LocalDate> dateDataType, ZoneId jvmZone, ZoneId someZone)
-    {
         LocalDate dateOfLocalTimeChangeForwardAtMidnightInJvmZone = LocalDate.of(1970, 1, 1);
         verify(jvmZone.getRules().getValidOffsets(dateOfLocalTimeChangeForwardAtMidnightInJvmZone.atStartOfDay()).isEmpty());
-
         LocalDate dateOfLocalTimeChangeForwardAtMidnightInSomeZone = LocalDate.of(1983, 4, 1);
         verify(someZone.getRules().getValidOffsets(dateOfLocalTimeChangeForwardAtMidnightInSomeZone.atStartOfDay()).isEmpty());
         LocalDate dateOfLocalTimeChangeBackwardAtMidnightInSomeZone = LocalDate.of(1983, 10, 1);
         verify(someZone.getRules().getValidOffsets(dateOfLocalTimeChangeBackwardAtMidnightInSomeZone.atStartOfDay().minusMinutes(1)).size() == 2);
 
-        return DataTypeTest.create()
-                .addRoundTrip(dateDataType, LocalDate.of(1952, 4, 3)) // before epoch
-                .addRoundTrip(dateDataType, LocalDate.of(1970, 1, 1))
-                .addRoundTrip(dateDataType, LocalDate.of(1970, 2, 3))
-                .addRoundTrip(dateDataType, LocalDate.of(2017, 7, 1)) // summer on northern hemisphere (possible DST)
-                .addRoundTrip(dateDataType, LocalDate.of(2017, 1, 1)) // winter on northern hemisphere (possible DST on southern hemisphere)
-                .addRoundTrip(dateDataType, dateOfLocalTimeChangeForwardAtMidnightInJvmZone)
-                .addRoundTrip(dateDataType, dateOfLocalTimeChangeForwardAtMidnightInSomeZone)
-                .addRoundTrip(dateDataType, dateOfLocalTimeChangeBackwardAtMidnightInSomeZone);
+        for (String timeZoneId : ImmutableList.of(UTC_KEY.getId(), jvmZone.getId(), someZone.getId())) {
```

Comment:
This loop works but for future reference take a look at `PostgresqlTypeMappingTest` to see how it uses `DataProvider`. The benefit is that if a test fails you can know exactly which timezone it failed for rather than having to guess which iteration of the loop it failed in.

No change requested since it's pre-existing. Can be a follow up.